app/controllers/api/v1/questions_controller.rb

```diff
@@ -0,0 +1,136 @@
+class Api::V1::QuestionsController < ApplicationController
+  
+  # Index method returns the list of questions JSON object
+  # GET on /questions
+  def index
+    @questions = Question.order(:id)
+    render json: @questions, status: :ok
+  end
+
+  # Show method returns the question object with id - {:id}
+  # GET on /questions/:id
+  def show
+    begin
+      @question = Question.find(params[:id])
+      render json: @question, status: :ok
+    rescue ActiveRecord::RecordNotFound
+      render json: $ERROR_INFO.to_s, status: :not_found
+    end
+  end
+
+  # Create method returns the JSON object of created question
+  # POST on /questions
+  def create
+    begin
+      questionnaire_id = params[:questionnaire_id] unless params[:questionnaire_id].nil?
+      num_of_existed_questions = Questionnaire.find(questionnaire_id).questions.size
+      question = Question.create(
+        txt: params[:txt],
+        questionnaire_id: questionnaire_id,
+        seq: num_of_existed_questions + 1,
+        question_type: params[:question_type],
+        break_before: true)
+      case question.question_type
+        when 'Scale'
+          question.weight = params[:weight]
+          question.max_label = 'Strongly agree'
+          question.min_label = 'Strongly disagree'
+        when 'Cake', 'Criterion'
+          question.weight = params[:weight]
+          question.max_label = 'Strongly agree'
+          question.min_label = 'Strongly disagree'
+          question.size = '50, 3'
+        when 'Dropdown'
+          question.alternatives = '0|1|2|3|4|5'
+          question.size = nil
+        when 'TextArea'
+          question.size = '60, 5'
+        when 'TextField'
+          question.size = '30'
+      end
+    
+      question.save!
+      render json: question, status: :created
+    rescue ActiveRecord::RecordNotFound
+      render json: $ERROR_INFO.to_s, status: :not_found and return  
+    rescue ActiveRecord::RecordInvalid
+      render json: $ERROR_INFO.to_s, status: :unprocessable_entity
+    end
+  end
+
+  # Destroy method deletes the question with id - {:id}
+  # DELETE on /questions/:id
+  def destroy
+    begin
+      @question = Question.find(params[:id])
+      @question.destroy
+    rescue ActiveRecord::RecordNotFound
+      render json: $ERROR_INFO.to_s, status: :not_found and return
+    rescue ActiveRecord::RecordInvalid
```

Comment:
I'm uncertain about when we might receive an Invalid Record error.

Reply:
Removed the rescue statement as it is not needed.

Reply:
Commit e86e810 addresses this review.